Add Kafka Tool.app Latest

OndroNR opened the pull request:
Uses install4j. Requires Java 8.

vitorgalvao commented:
There is trailing whitespace that needs to be removed.

vitorgalvao commented on Casks/kafka-tool.rb:
```diff
@@ -0,0 +1,19 @@
+  license :commercial
+
+  app 'Kafka Tool.app'
+  preflight do
```
Please add a blank line before.

OndroNR commented:
Just force pushed update.

vitorgalvao commented on Casks/kafka-tool.rb:
```diff
@@ -0,0 +1,20 @@
+  app 'Kafka Tool.app'
+
+  preflight do
+    system "#{staged_path}/Kafka Tool Installer.app/Contents/MacOS/JavaApplicationStub", '-q', '-dir', "#{staged_path}"
```
Why are you running this uninstaller and linking an `app`?

OndroNR replied:
I am running install4j installer in quiet mode which installs Kafka Tool.app. I am new to both brew cask and OSX ecosystem, I might get something wrong. Without the `app`, Kafka Tool does not appear in Applications. Is there any problem?

vitorgalvao replied:
> Without the `app`, Kafka Tool does not appear in Applications.

That is unlikely. I won’t have my computer for a while so can’t test everything myself, but `app` simply links something that already exists. I this case, only the installer exists, so `app` is doing nothing and should be removed.

In addition, this requires [`uninstall`](https://github.com/caskroom/homebrew-cask/blob/master/doc/CASK_LANGUAGE_REFERENCE.md#uninstall-stanza-details).

OndroNR replied:
I guess that Kafka Tool is installed in `preflight` which runs before linking app. Kafka Tool is installed into `staged_path`. By default, installer install into `/Applications`.

jawshooah replied:
There's a bit of inconsistency with how these installers are applied. [`dendroscope`](https://github.com/caskroom/homebrew-cask/blob/dee5d6c63e9b132dcdf4d50446680195ec5ce162/Casks/dendroscope.rb), for example, installs the app to `staged_path` and links it with the `app` stanza. [`thinkorswim`](https://github.com/caskroom/homebrew-cask/blob/7fb617c0b96ebdaa932f76c118d518b4ba193c7a/Casks/thinkorswim.rb), however, just installs the app to `/Applications`.

@vitorgalvao, I'd say that with the planned changes to installation behavior across all casks, the latter behavior should be preferred for now. @OndroNR, please follow the `thinkorswim` example and use `installer :script` rather than `preflight` and `app`.

vitorgalvao replied:
Agreed.

OndroNR commented:
Thanks for guidance @jawshooah. I have used installer :script without sudo. Kafka Tool gets installed into `/opt/homebrew-cask/Caskroom/kafka-tool/latest`, I guess that was workdir for installer which could not access `/Applications`. App does not appear in Launchpad, nor in Applications. But I can find it in Spotlight. `thinkorswim` gets installed to `/opt/homebrew-cask/Caskroom/thinkorswim/latest`, similarly as Kafka Tool, not in `/Applications`. It also appears in Launchpad, but in Applications it shows as folder with binary, JARs and some other files. I don't know what to do now, please help.

```
➜  homebrew-cask git:(kafka-tool) brew cask install kafka-tool
==> Caveats
kafka-tool requires Java 8+, you can install the latest Java using
  brew cask install java

==> Downloading http://www.kafkatool.com/download/kafkatool.dmg
Already downloaded: /Library/Caches/Homebrew/kafka-tool-latest.dmg
==> Running installer script Kafka Tool Installer.app/Contents/MacOS/JavaApplicationStub
==> The installation directory has been set to /opt/homebrew-cask/Caskroom/kafka-tool/latest.
==> Extracting files ...
==> Finishing installation ...
🍺  kafka-tool staged at '/opt/homebrew-cask/Caskroom/kafka-tool/latest' (217 files, 225M)
```

jawshooah commented:
Interesting, `thinkorswim` installs to `/Applications` just fine for me:

```
$ brew cask install thinkorswim
==> Downloading https://mediaserver.thinkorswim.com/installer/InstFiles/thinkorswim_installer.dmg
Already downloaded: /Library/Caches/Homebrew/thinkorswim-latest.dmg
==> Running installer script thinkorswim Installer.app/Contents/MacOS/JavaApplicationStub
==> The installation directory has been set to /Applications/thinkorswim.
==> Extracting files ...
==> Finishing installation ...
🍺  thinkorswim staged at '/opt/homebrew-cask/Caskroom/thinkorswim/latest' (22 files, 39M)
```

So does `kafka-tool`:

```
$ brew cask install kafka-tool
==> Caveats
kafka-tool requires Java 8+, you can install the latest Java using
  brew cask install java

==> Downloading http://www.kafkatool.com/download/kafkatool.dmg
Already downloaded: /Library/Caches/Homebrew/kafka-tool-latest.dmg
==> Running installer script Kafka Tool Installer.app/Contents/MacOS/JavaApplicationStub
==> The installation directory has been set to /Applications.
==> Extracting files ...
==> Finishing installation ...
🍺  kafka-tool staged at '/opt/homebrew-cask/Caskroom/kafka-tool/latest' (19 files, 51M)
```

What OS version are you using? What are the permissions on your `/Applications`? Here's mine:

```
$ ls -ld /Applications/
drwxrwxr-x+ 155 root admin 5270 Oct  8 21:41 /Applications/
```

jawshooah commented:
Also, since we're using an `installer`, we need an `uninstall` stanza. Looks like all that's required is `uninstall :delete => '/Applications/Kafka Tool.app'

jawshooah commented:
Looks like `zap :delete => '~/.kafkatool'` should be added as well.

OndroNR commented:
My permissions looks same.
```
➜  ~  ls -ld /Applications/
drwxrwxr-x+ 64 root  admin  2176 Oct  9 01:16 /Applications/
```

OndroNR commented:
What should I do? Force install it into `/Applications` ?

OndroNR commented:
@jawshooah ping

jawshooah commented:
I'm not sure why it's installing to a different directory for you. Works fine for me. Regardless, my points above regarding `uninstall` (and optionally `zap`) need to be addressed before this can be merged.

jawshooah commented:
Is your user a member of the `admin` group? If not, you likely don't have write permissions inside `/Applications`. You'll have to either add your user to the `admin` group or change the ownership of `/Applications`.

OndroNR commented:
I am only user and I am member of `adminz group.

OndroNR commented:
I have removed `:sudo => false` and it now requires password, and installs into `/Applications`. I also added `uninstall :delete` stanza. I would rather not delete settings (4 small XML files) if it is not required. I have squashed all commits. Anything else, I should do? Thanks

vitorgalvao commented:
> I would rather not delete settings (4 small XML files) if it is not required.

That is what [`zap`](https://github.com/caskroom/homebrew-cask/blob/master/doc/CASK_LANGUAGE_REFERENCE.md#zap-stanza-details) is for.

OndroNR commented:
Great, just added `zap`.

vitorgalvao commented:
Thank you for the contribution. It was merged directly as 26a888b6683d3bea66a14b40228af514ee723206 to keep commit history cleaner. Your contribution is still credited to you.

OndroNR commented:
Thanks
